packages/flutter/lib/src/material/progress_indicator.dart

```diff
@@ -243,6 +243,196 @@ class _LinearProgressIndicatorPainter extends CustomPainter {
+///           value: 0.5,
+///           semanticsLabel: 'Linear progress indicator',
+///         ),
+///       ],
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// This example shows a [LinearProgressIndicator] with a changing color.
+///
+/// ```dart
+/// AnimationController controller;
+/// Animation<Color> animation;
+///
+/// @override
+/// void initState() {
+///   controller = AnimationController(
+///     vsync: this,
+///     duration: const Duration(seconds: 5),
+///   );
+///   animation = ColorTween(
+///     begin: Colors.green,
+///     end: Colors.blue,
+///   ).animate(controller);
+///   controller.repeat(reverse: true);
+///   super.initState();
+/// }
+///
+/// @override
+/// void dispose() {
+///   controller.dispose();
+///   super.dispose();
+/// }
+///
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Padding(
+///       padding: const EdgeInsets.all(20.0),
+///       child: Column(
+///         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+///         children: [
+///           Text(
+///             'Linear progress indicator with a fixed color',
+///             style: const TextStyle(fontSize: 20),
+///           ),
+///           LinearProgressIndicator(
+///             valueColor: animation,
+///             semanticsLabel: 'Linear progress indicator',
+///           ),
+///         ],
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material_ticker}
+///
+/// This example shows a [LinearProgressIndicator] with a changing value.
+///
+/// ```dart
+/// AnimationController controller;
+/// Animation<double> animation;
+///
+/// @override
+/// void initState() {
+///   controller = AnimationController(
+///     vsync: this,
+///     duration: const Duration(seconds: 5),
+///   );
+///   animation = Tween<double>(
+///     begin: 0,
+///     end: 1,
+///   ).animate(controller)
+///     ..addListener(() {
+///       setState(() {});
+///     });
+///   controller.repeat(reverse: true);
+///   super.initState();
+/// }
+///
+/// @override
+/// void dispose() {
+///   controller.dispose();
+///   super.dispose();
+/// }
+///
+/// Widget build(BuildContext context) {
+///   return Scaffold(
+///     body: Padding(
+///       padding: const EdgeInsets.all(20.0),
+///       child: Column(
+///         mainAxisAlignment: MainAxisAlignment.spaceEvenly,
+///         children: [
+///           Text(
+///             'Linear progress indicator with a fixed color',
+///             style: const TextStyle(fontSize: 20),
+///           ),
+///           LinearProgressIndicator(
+///             value: animation.value,
+///             semanticsLabel: 'Linear progress indicator',
+///           ),
+///         ],
+///       ),
+///     ),
+///   );
+/// }
+/// ```
+/// {@end-tool}
```

Comment:
These samples are great, but could you combine them into a single dartpad sample with a fixed value and color?

Reply:
Alright, I'll combine them